specs/interop.md

```diff
@@ -0,0 +1,396 @@
+saying it is that the dependency set defines the set of initiating messages that are valid for a relaying
+message to be included. A relaying message MUST have an initiating message that is included in a chain
+in the dependency set.
+
+The dependency set is defined by chain id. Since it is impossible to enforce uniqueness of chain ids,
+social consensus MUST be used to determine the chain that represents the canonical chain id. This
+particularly impacts the block builder as they SHOULD use the chain id to assist in validation
+of relaying messages.
+
+### Chain ID
+
+The concept of a chain id was introduced in [EIP-155](https://eips.ethereum.org/EIPS/eip-155) to prevent
+replay attacks between chains. This EIP does not specify the max size of a chain id, although
+[EIP-2294](https://eips.ethereum.org/EIPS/eip-2294) attempts to add a maximum size. Since this EIP is
+stagnant, all representations of chain ids MUST be the `uint256` type.
+
+## Interop Network Upgrade
+
+The interop network upgrade timestamp defines the timestamp at which all functionality in this document is considered
+the consensus rules for an OP Stack based network. On the interop network upgrade block, a set of deposit transaction
+based upgrade transactions are deterministically generated by the derivation pipeline in the following order:
+
+- L1 Attributes Transaction calling `setL1BlockValuesEcotone`
+- User deposits from L1
+- Network Upgrade Transactions
+  - L1Block deployment
+  - CrossL2Inbox deployment
+  - CrossL2Output deployment
+  - Update L1Block Proxy ERC-1967 Implementation Slot
+  - Update CrossL2Inbox Proxy ERC-1967 Implementation Slot
+  - Update CrossL2Output Proxy ERC-1967 Implementation Slot
+  - Mint Cross Chain Ether Liquidity in CrossL2Inbox
+
+The execution payload MUST set `noTxPool` to `false` for this block.
+
+The exact definitions for these upgrade transactions are still to be defined.
+
+## Derivation Pipeline
+
+The derivation pipeline enforces invariants on safe blocks that include relaying transactions.
+
+- The relaying message MUST have a corresponding initiating message
+- The initiating message that corresponds to a relaying message MUST come from an allowed chain (dependency set)
+
+Blocks that contain transactions that relay cross domain messages to the destination chain where the
+initiating transaction does not exist MUST be considered invalid and MUST not be allowed by the
+derivation pipeline to be considered safe.
+
+There is no concept of replay protection within the protocol since there is no replay protection
+mechanism that fits well for all applications. Users MAY submit an arbitrary number of relaying
+messages per initiating message. This greatly simplifies the protocol and allows for
+applications to build application specific replay protection.
+
+### Depositing a Relaying Message
+
+Deposit transactions (force inclusion transactions) give censorship resistance to layer two networks.
```

Comment:
Need to add explicit call out the the deposit is dropped by the derivation pipeline

Reply:
This has been completed